src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletState.scala

```diff
@@ -135,24 +140,59 @@ object ErgoWalletState {
   val noWalletFilter: FilterFn = (_: TrackedBox) => true
 
   def initial(ergoSettings: ErgoSettings, parameters: Parameters): Try[ErgoWalletState] = {
-    WalletRegistry.apply(ergoSettings).map { registry =>
+    WalletRegistry.apply(ergoSettings).flatMap { registry =>
       val ergoStorage: WalletStorage = WalletStorage.readOrCreate(ergoSettings)(ergoSettings.addressEncoder)
       val offChainRegistry = OffChainRegistry.init(registry)
       val walletVars = WalletVars.apply(ergoStorage, ergoSettings)
       val maxInputsToUse = ergoSettings.walletSettings.maxInputs
-      ErgoWalletState(
-        ergoStorage,
-        secretStorageOpt = None,
-        registry,
-        offChainRegistry,
-        outputsFilter = None,
-        walletVars,
-        stateReaderOpt = None,
-        mempoolReaderOpt = None,
-        utxoStateReaderOpt = None,
-        parameters,
-        maxInputsToUse
-      )
+
+      def buildState =
+        ErgoWalletState(
+          ergoStorage,
+          secretStorageOpt = None,
+          registry,
+          offChainRegistry,
+          outputsFilter = None,
+          walletVars,
+          stateReaderOpt = None,
+          mempoolReaderOpt = None,
+          utxoStateReaderOpt = None,
+          parameters,
+          maxInputsToUse
+        )
+
+      // compatibility bridge to version 4.0.26 where ScanId changed from Short to Int, let's migrate them
+      if (Version(ergoSettings.scorexSettings.network.appVersion).compare(Version(4, 0, 26)) >= 0) {
+        val legacyScans = ergoStorage.getLegacyScans
+        if (legacyScans.exists(_.scanId.intValue < 0)) {
+          log.info(s"Scans already migrated")
+          Success(buildState)
```

Comment:
@kushti This is the only problematic part, after migrating scanIds (deleting all scans with `Short` keys and creating new scan with `Int`  keys), `getLegacyScans` returns some weird negative `Short` key as if the `remove` operation has BUG in it.